telegram/ext/_application.py

```diff
@@ -297,14 +298,10 @@ def __init__(
         self.post_stop: Optional[
             Callable[["Application[BT, CCT, UD, CD, BD, JQ]"], Coroutine[Any, Any, None]]
         ] = post_stop
-
-        if isinstance(concurrent_updates, int) and concurrent_updates < 0:
-            raise ValueError("`concurrent_updates` must be a non-negative integer!")
-        if concurrent_updates is True:
-            concurrent_updates = 256
-        self._concurrent_updates_sem = asyncio.BoundedSemaphore(concurrent_updates or 1)
-        self._concurrent_updates: int = concurrent_updates or 0
-
+        self._concurrent_updates_sem = asyncio.BoundedSemaphore(
+            update_processor.max_concurrent_updates or 1
+        )
```

Comment:
We no longer need this semaphore, as `BaseUpdateProcossor.do_process_updates` now has a semaphore :)